urbansearch/server/main.py

```diff
@@ -23,6 +24,12 @@ def __init__(self, run=True):
         self.app = Flask(__name__)
         self.register_blueprints()
 
+        @self.app.after_request
+        def apply_caching(response):
+            response.headers["Access-Control-Allow-Origin"] = "*"
+            response.headers["Access-Control-Allow-Headers"] = "Content-Type"
+            return response
+
         if run:
```

Comment:
The Server class should be used to create a new server running the server should be handled in a seperate file (eg. wsgi.py)